tests/integration/test_scan.py

```diff
@@ -131,6 +131,41 @@ def _run_tests(op, n0, shape, dt, axis, out0, outtype):
 ]
 
 
+@pytest.mark.parametrize(
+    "dtype, outtype",
+    [
+        (np.int16, np.float64),
+        (np.complex64, np.float64),
+        (np.float32, np.int64),
+    ],
+    ids=str,
+)
+@pytest.mark.parametrize(
+    "op",
+    [
+        pytest.param("cumsum", marks=pytest.mark.xfail),
+        "cumprod",
+        "nancumsum",
+        "nancumprod",
+    ],
+    ids=str,
+)
+def test_scan_out_dtype_mismatch(dtype, outtype, op):
+    # When CUNUMERIC_TEST=1:
+    # out_np: array([0., 0., 0., 0., 0., 0.])
+    # out_num:
+    # array([6.8983227e-310, 6.8983227e-310, 6.8983227e-310, 6.8983227e-310,
+    #        6.8983227e-310, 6.8983227e-310])
+    shape = (1, 2, 3)
+    out_shape = (6,)
+    arr = np.empty(shape)
```

Comment:
`empty` is a good choice for out-arrays, I might consider something more defined and repeatable for inputs

Reply:
Good point, actually found more issues after changing the input.